src/include/common/container/chunked_array.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+namespace noisepage::common {
+
+/**
+ * Describes a chunked array. A chunked array batch allocates Size number of elements
+ * of type Type at a given time. The premise for this is to amortize comparably
+ * expensive memory allocation calls over a large number of inserts.
+ */
```

Comment:
The issue I was trying to avoid with `std::vector` is that its iterator can get invalidated when adding new elements to the vector. `ChunkedArray` ensures that the iterator is still valid when new elements are added.

Reply:
I see! Can you add that comment here since one may wonder?

Reply:
Added as a comment.